### PR TITLE
Add --version command and version metadata

### PR DESCRIPTION
**Pull request by MarkRx**
### **User description**
Implements #1370

Checks version info from pyproject.toml if running locally* before checking for pip metadata.

* Uses tomlib which was added in 3.11. Will return "unknown" if using 3.10. The other option would be to bump the minimum version to 3.11 or else import a 3rd party toml library

___

### **PR Type**
enhancement

___

### **Description**
- Added a `get_version` function to retrieve version information from `pyproject.toml` or pip metadata.
- Implemented a `--version` command in the CLI to display the current version of the application.
- Enhanced logging by including version metadata in the tags for better traceability.
- Updated the project version in `pyproject.toml` to 0.2.6.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>litellm_ai_handler.py</strong><dd><code>Add version metadata to logging tags</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/ai_handlers/litellm_ai_handler.py

<li>Imported <code>get_version</code> from <code>pr_agent.version</code>.<br> <li> Added version metadata to tags in <code>capture_logs</code> function.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1384/files#diff-ea1acaa0907f3410665530fbc4cda2ab524de2772e0bbe10bad4648b8be35dfe">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>cli.py</strong><dd><code>Implement --version command in CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/cli.py

<li>Imported <code>get_version</code> from <code>pr_agent.version</code>.<br> <li> Added <code>--version</code> command to CLI.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1384/files#diff-88577a9bac5c261b52dcec31fdf5e525cef5c6ddffdcb6bde0e9ba183fa05996">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>version.py</strong><dd><code>Add version retrieval utility</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/version.py

<li>Created <code>get_version</code> function to retrieve version from <code>pyproject.toml</code> <br>or pip metadata.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1384/files#diff-15d4b05ca33c2fdef7004673bcc9519abdf897feb9e6072a45e3ff869002ac80">+20/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Update project version in pyproject.toml</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pyproject.toml

- Updated project version to 0.2.6.

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1384/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "pr-agent"
-version = "0.2.4"
+version = "0.2.6"
```
Comment (mrT23) [inline pyproject.toml:7]: the current version is 0.2.5

File: pr_agent/version.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                return data["project"]["version"]
+        else:
+            get_logger().error("Unable to determine local version from pyproject.toml")
+
+    # Otherwise get the installed pip package version
+    try:
+        return version('pr-agent')
+    except PackageNotFoundError:
+        get_logger().error("Unable to find package named 'pr-agent'")
+        return "unknown"
```
Comment (mrT23) [inline pr_agent/version.py:23]: add this to an existing `util` file. no need for new file

**Comment by mrT23** [comment]
looks good. after fixing the feedback i gave, can be merged

File: pr_agent/version.py (hunk 2 above)
Comment (MarkRx) [reply pr_agent/version.py:23]: Which one should it go in?

Comment (mrT23) [reply pr_agent/version.py:23]: https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/algo/utils.py

**Review by mrT23 — APPROVED** [review]

**Pull request merged**